app/master/job_config.py

```diff
@@ -18,3 +35,119 @@ def __init__(self, name, setup_build, teardown_build, command, atomizer, max_exe
         self.atomizer = atomizer
         self.max_executors = max_executors
         self.max_executors_per_slave = max_executors_per_slave
+
+    @classmethod
+    def construct_from_dict(cls, name, config_dict):
```

Comment:
How about just put all the logic in `__init__`? Is there some other places that JobConfig is instantiated directly without using this factory method?